crates/storage/libmdbx-rs/src/error.rs

```diff
@@ -1,70 +1,116 @@
 use libc::c_int;
-use std::{ffi::CStr, fmt, result, str};
+use std::result;
 
 /// An MDBX result.
 pub type Result<T> = result::Result<T, Error>;
 
 /// An MDBX error kind.
 #[derive(Debug, thiserror::Error, Clone, PartialEq, Eq)]
 pub enum Error {
-    /// The key/value pair already exists.
+    /// Key/data pair already exists.
+    #[error("key/data pair already exists")]
     KeyExist,
-    /// The requested key/value pair was not found.
+    /// No matching key/data pair found.
+    #[error("no matching key/data pair found")]
     NotFound,
+    /// The cursor is already at the end of data.
+    #[error("the cursor is already at the end of data")]
     NoData,
-    /// The requested page was not found.
+    /// Requested page not found.
+    #[error("requested page not found")]
     PageNotFound,
-    /// The database is corrupted (e.g. a page was a wrong type)
+    /// Database is corrupted.
+    #[error("database is corrupted")]
     Corrupted,
-    /// The environment had a fatal error (e.g. failed to update a meta page)
+    /// Fatal environment error.
+    #[error("fatal environment error")]
     Panic,
+    /// DB version mismatch.
+    #[error("DB version mismatch")]
     VersionMismatch,
-    /// File is not a valid MDBX file
+    /// File is not an MDBX file.
+    #[error("file is not an MDBX file")]
     Invalid,
-    /// Environment map size reached.
+    /// Environment map size limit reached.
+    #[error("environment map size limit reached")]
     MapFull,
-    /// Environment reached the maximum number of databases.
+    /// Too many DBI-handles (maxdbs reached).
+    #[error("too many DBI-handles (maxdbs reached)")]
     DbsFull,
-    /// Environment reached the maximum number of readers.
+    /// Too many readers (maxreaders reached).
+    #[error("too many readers (maxreaders reached)")]
     ReadersFull,
-    /// The transaction has too many dirty pages (i.e. the transaction is too big).
+    /// Transaction has too many dirty pages (i.e., the transaction is too big).
+    #[error("transaction has too many dirty pages (i.e., the transaction is too big)")]
     TxnFull,
-    /// The cursor stack is too deep.
+    /// Cursor stack limit reached.
+    #[error("cursor stack limit reached")]
     CursorFull,
-    /// The page does not have enough space.
+    /// Page has no more space.
+    #[error("page has no more space")]
     PageFull,
-    /// The database engine was unable to extend mapping, e.g. the address space is unavailable or
-    /// busy.
-    ///
-    /// This can mean:
-    /// - The database size was extended by other processes beyond the environment map size, and
-    ///   the engine was unable to extend the mapping while starting a read transaction. The
-    ///   environment should be re-opened to continue.
-    /// - The engine was unable to extend the mapping during a write transaction or an explicit
```

Comment:
can we keep this comment for this variant?

Reply:
Sure I've just added it back